### ISSUE TITLE
Q: in weave as k8s add-on, what is the equivalent of weave cli commands?

### ISSUE THREAD

**Issue by aalba6675** [weaveworks/weave#3052]
When using weave as an add-on to kubernetes, how do you inspect the state of weave, or what is the equivalent of the weave CLI commands like `weave ipam status`?

I see that every node has a weave-net-* pod running, obviously, there are no node (systemd)services or binaries installed.

In this situation, what is the recommend best practice to manage weave when most of it is hidden in the pods?

If I do something like running weave from inside the pod, it doesn't work.
```
kubectl exec weave-net-0lhn5 -n kube-system -- /home/weave/weave status ipam
Defaulting container name to weave.
Use 'kubectl describe pod/weave-net-0lhn5' to see all of the containers in this pod.
/home/weave/weave: line 250: docker: not found
ERROR: Unable to parse docker version
```

…though kube-peers works
```
kubectl exec weave-net-0lhn5 -n kube-system -- /home/weave/kube-peers
Defaulting container name to weave.
Use 'kubectl describe pod/weave-net-0lhn5' to see all of the containers in this pod.
192.168.1.10
192.168.1.11
192.168.1.12
192.168.1.13
```

**Comment by bboreham** [comment]
@aalba6675 you need to add `--local` to tell it not to go via Docker - see #2603 

Could you expand a bit on what kinds of things you want to do "to manage weave"?

#2603 covers the `status` command specifically.
#2612 is about troubleshooting.
#2797 is about removing deleted nodes.

When I try to think of a simple way to present this, a lot of complexities come to mind - e.g. if you have 10 nodes, do you want to see status from all of them?  Is it useful to pick one at random?

So, I'd be really interested if you could walk through some scenarios, help to generate some ideas about how this could work.

**Comment by aalba6675** [comment]
@bboreham  I'm groping here, but generally for SDN add-ons (not just weave mind you) there is a disconnect between using it as add-ons and externally configured SDN with regards to documentation. Once it used as an add-on, it seems much more mysterious and opaque — don't know if that makes much sense to you — but coming from the angle of operator training and understanding viz a viz an external SDN.

Weave recommends using the add-on for k8s, but what if I want to have a "standard" weave for external services or just non-k8s workloads: in one use case, we want to have the k8s etcd cluster as a set container "pets" but also trying not to introduce another SDN.  Do you recommend separate weave enclaves and figure out routing some how, rather than "one-SDN-to-rule-them-all"?

It would be good if you address this somehow: best practice for integrating externally configured weave nets with add-on weave nets. Sorry for the rambling...

**Comment by bboreham** [comment]
The original question is resolved by #3055, and the docs are at https://www.weave.works/docs/net/latest/kube-addon/#operations

**Comment by bboreham** [comment]
> Weave recommends using the add-on for k8s, but what if I want to have a "standard" weave for external services or just non-k8s workloads: in one use case, we want to have the k8s etcd and workload datastores and message queues (consul, redis, etcd, rabbitmq) as sets of external container "pets" but also trying not to introduce another SDN. Do you recommend separate weave enclaves and figure out routing somehow, rather than "one-SDN-to-rule-them-all"?

If you want Kubernetes and non-Kubernetes containers to talk to each other, we recommend connecting them all to the same Weave network.  Simplest is to join more peers on to the existing Kubernetes Weave network, but if you want to go the other way round then add `--ipalloc-init=observer` into the launch command of the Kubernetes peers so they won't try to form their own network.
